### DISCUSSION TITLE
Replace decimal separators

### DISCUSSION THREAD

**Discussion by terjene** [JoshClose/CsvHelper#2135] [Q&A] [answered]
I'm importing csv files with different formats, and also not the same columns every time..
The decimal separator can be either ',' or '.'
I've kind of solved this with the below ClassMap, but is this the best way?
Also, if the column SalesPrice is missing from the csv, I get an null exception, when trying to decimal.parse.

Any better way to do this? (I guess so...)

```cs
public SystemImportClassMap()
{
	Map(m => m.LineNumber).Convert(record => record.Row.Parser.Row);
	Map(m => m.CostPrice).Convert(record => decimal.Parse(record.Row["CostPrice"].ToString().Replace('.', ','))).Name("CostPrice");
	Map(m => m.SalesPrice).Convert(record => decimal.Parse(record.Row["SalesPrice"].ToString().Replace('.', ','))).Name("SalesPrice");
	Map(m => m.Memo).Name("Memo");
}
```

**Comment by JoshClose** [comment, marked as answer]
That may be a little slow. I would suggest finding what culture the CSV is in, then setting that. There could be other things in the file that are different for that culture also. Cultures that use `,` for decimal may also use `.` for thousands separator.
